### ISSUE TITLE
Unable to build waterway

### ISSUE THREAD

**Issue by frankystone** [widelands/widelands#3880]
In this screenshot i can build waterways at all flags, except the second from above.
![no_waterway](https://user-images.githubusercontent.com/6730556/80303736-55b3da00-87b2-11ea-835d-414320b959b3.jpg)

If you load the savegame and scroll to the west there is another coastline were waterways could be build on every flag, except from the flag of the port.
[no_waterways.wgf.zip](https://github.com/widelands/widelands/files/4535034/no_waterways.wgf.zip)

Don't know if this is really a bug, anyway it is not clear why waterways couldn't be build from these flags.

Widelands version: 5d8cc18@shipping-tweaks-b21
OS: archlinux

**Comment by Noordfrees** [comment]
I cannot reproduce, the buildWW button is shown for every flag on the coastline for me. Which flag (coords) do you mean?

**Comment by frankystone** [comment]
Yes, the button is availbale but if you click on it the blue waterway markers are missing. Also clicking on a flag where the waterway should go to prompts you with the action window to cancel building a waterway.

Sorry, can't provide coordinates (release build): In the picture shown above there is a waterway going from one flag. From there it is the next flag in south/east (near the port).

**Comment by Noordfrees** [comment]
Ah! The places where you'd like to build the waterway are blocked by the PortDock. This is an invisible immovable that is automatically created by a port-type warehouse and it's the location where ships have to sail to interact with the port. Since waterways cannot be built over any immovables it's not possible to build a waterway there. So the behaviour is correct.
Perhaps we should mention that WWs can't be built directly in front of a port in the seafaring tutorial?

**Comment by frankystone** [comment]
> Perhaps we should mention that WWs can't be built directly in front of a port in the seafaring tutorial?

+1

Additionally disable the button (don't show it) at this places?

**Comment by Noordfrees** [comment]
> Additionally disable the button (don't show it) at this places?

We also show the build road button at flags if all surrounding nodes are blocked by immovables, so -1

**Comment by frankystone** [comment]
> > Additionally disable the button (don't show it) at this places?
> 
> We also show the build road button at flags if all surrounding nodes are blocked by immovables, so -1

I think this is a different thing: One can visually see that roads might not be buildable on those spots (it can be seen that immovables block roads, of course not all immovables though), whereas in regard with waterways one can't see anything.

**Comment by Noordfrees** [comment]
Another idea would be to graphically indicate the port docks, e.g. by drawing *Kaimauern* or some other port-typical stuff. Would be a bit more effort, but perhaps someone will have time for this…
